grafana-plugin/src/components/GForm/GForm.tsx

```diff
@@ -69,45 +102,55 @@ function renderFormControl(formItem: FormItem, register: any, control: any) {
   }
 }
 
-const GForm = (props: GFormProps) => {
-  const { data, form, onSubmit } = props;
-
-  const handleSubmit = useCallback(
-    (data: any) => {
-      const normalizedData = Object.keys(data).reduce((acc, key) => {
-        const formItem = form.fields.find((formItem) => formItem.name === key);
-
-        const value = formItem?.normalize ? formItem.normalize(data[key]) : nullNormalizer(data[key]);
-
-        return {
-          ...acc,
-          [key]: value,
-        };
-      }, {});
-
-      onSubmit(normalizedData);
-    },
-    [onSubmit]
-  );
-
-  return (
-    <Form maxWidth="none" id={form.name} defaultValues={data} onSubmit={handleSubmit}>
-      {({ register, errors, control }) => {
-        return form.fields.map((formItem: FormItem, formIndex: number) => (
-          <Field
-            key={formIndex}
-            disabled={formItem.getDisabled ? formItem.getDisabled(data) : false}
-            label={formItem.label || capitalCase(formItem.name)}
-            invalid={!!errors[formItem.name]}
-            error={`${capitalCase(formItem.name)} is required`}
-            description={formItem.description}
-          >
-            {renderFormControl(formItem, register, control)}
-          </Field>
-        ));
-      }}
-    </Form>
-  );
-};
+class GForm extends React.Component<GFormProps, {}> {
+  render() {
+    const { form, data } = this.props;
+
+    return (
+      <Form maxWidth="none" id={form.name} defaultValues={data} onSubmit={this.handleSubmit}>
+        {({ register, errors, control, getValues, setValue }) => {
+          return form.fields.map((formItem: FormItem, formIndex: number) => {
+            if (formItem.shouldShow && !formItem.shouldShow(getValues())) {
+              setValue(formItem.name, undefined); // clear input value on hide
```

Comment:
No other place where I can clear the value for it, I think this is totally fine.